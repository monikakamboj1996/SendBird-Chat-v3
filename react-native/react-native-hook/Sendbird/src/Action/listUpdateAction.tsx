export const ListUpdate = list => {
  return {
    type: 'List_Update',
    list: list,
  };
};
