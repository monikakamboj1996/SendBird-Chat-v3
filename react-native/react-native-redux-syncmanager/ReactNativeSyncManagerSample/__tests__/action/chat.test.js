import 'react-native';
import React from 'react';
import Action from '../../src/action/chat';

describe('action/chat', () => {
  it('selectMessage', () => {
    const result = Action.selectMessage({ message: 'test' });
    expect(result).toMatchSnapshot();
  });

  it('deselectAllMessages', () => {
    const result = Action.deselectAllMessages();
    expect(result).toMatchSnapshot();
  });

  it('addMessages', () => {
    const result = Action.addMessages([{ message: 'test' }, { message: 'test2' }]);
    expect(result).toMatchSnapshot();
  });

  it('updateMessages', () => {
    const result = Action.updateMessages([{ message: 'test' }, { message: 'test2' }]);
    expect(result).toMatchSnapshot();
  });

  it('removeMessages', () => {
    const result = Action.removeMessages([{ message: 'test' }, { message: 'test2' }]);
    expect(result).toMatchSnapshot();
  });

  it('clearMessages', () => {
    const result = Action.clearMessages();
    expect(result).toMatchSnapshot();
  });
});
