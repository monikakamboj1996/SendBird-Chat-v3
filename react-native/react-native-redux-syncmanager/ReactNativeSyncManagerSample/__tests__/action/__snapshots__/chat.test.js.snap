// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`action/chat addMessages 1`] = `
Object {
  "payload": Object {
    "messages": Array [
      Object {
        "message": "test",
      },
      Object {
        "message": "test2",
      },
    ],
  },
  "type": "message-add",
}
`;

exports[`action/chat clearMessages 1`] = `
Object {
  "type": "message-clear",
}
`;

exports[`action/chat deselectAllMessages 1`] = `
Object {
  "type": "message-deselect-all",
}
`;

exports[`action/chat removeMessages 1`] = `
Object {
  "payload": Object {
    "messages": Array [
      Object {
        "message": "test",
      },
      Object {
        "message": "test2",
      },
    ],
  },
  "type": "message-remove",
}
`;

exports[`action/chat selectMessage 1`] = `
Object {
  "payload": Object {
    "message": Object {
      "message": "test",
    },
  },
  "type": "message-select",
}
`;

exports[`action/chat updateMessages 1`] = `
Object {
  "payload": Object {
    "isTransformFailedToManual": undefined,
    "messages": Array [
      Object {
        "message": "test",
      },
      Object {
        "message": "test2",
      },
    ],
  },
  "type": "message-update",
}
`;
